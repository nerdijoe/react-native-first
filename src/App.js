import React from 'react'
import {
  View,
  Text,
  Image,
  StatusBar
} from 'react-native'
import { Router, Scene } from 'react-native-router-flux';
import { connect } from 'react-redux'

import { actionFetchData, actionFetchDataArenas } from './actions'
import FadeInView from './components/FadeInView'
import CardList from './components/CardList'
import CardListV2 from './components/CardListV2'
import Footer from './components/Footer'
import Navbar from './components/Navbar'
import ArenaList from './components/ArenaList'
import CardDetail from './components/CardDetail'
import ArenaDetail from './components/ArenaDetail'
import CamCam from './components/CamCam'

const styles = {
  barTitle: {
    backgroundColor: '#8EF6E4',

  }
}

export class App extends React.Component {
  componentDidMount() {
    this.props.actionFetchData()
    this.props.actionFetchDataArenas()
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <StatusBar
          backgroundColor="#F6F6F6"
          barStyle="dark-content"
        />
        <FadeInView style={{ width: '100%', height: '25%', backgroundColor: 'powderblue'}}>
            <Image
              source={{uri: 'http://descargarclashroyalegratis.net/wp-content/uploads/2016/04/clash-min.jpg'}}
              style={{width:1200/2.9, height: 630/2.9}}
            />
        </FadeInView>

        <Navbar />

        <Router>
           <Scene key="cards" component={CardList} title="Cards"  navigationBarStyle={styles.barTitle} />
           <Scene key="cards2" component={CardListV2} title="Cards with ListView" initial={true} navigationBarStyle={styles.barTitle} />
           <Scene key="arenas" component={ArenaList} title="Arenas" navigationBarStyle={styles.barTitle} />
           <Scene key="cdetail" component={CardDetail} title="Card Detail" navigationBarStyle={styles.barTitle} />
           <Scene key="adetail" component={ArenaDetail} title="Arena Detail" navigationBarStyle={styles.barTitle} />

           <Scene key="camcam" component={CamCam} title="CamCam" navigationBarStyle={styles.barTitle} />


       </Router>

        <Footer />
      </View>
    )
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    actionFetchData: () => { dispatch(actionFetchData()) },
    actionFetchDataArenas: () => { dispatch(actionFetchDataArenas()) }
  }
}

const connectedApp = connect(null, mapDispatchToProps)(App)
export default connectedApp
