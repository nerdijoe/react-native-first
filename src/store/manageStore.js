import { createStore, applyMiddleware, compose } from 'redux'
import logger from 'redux-logger'
import thunk from 'redux-thunk'

import clasApp from '../reducers'

const middlewares = applyMiddleware( logger, thunk )

const store = createStore(clasApp, middlewares)

export default store
