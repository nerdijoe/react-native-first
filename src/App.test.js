import renderer from 'react-test-renderer'
import React from 'react'
import ReactDOM from 'react-dom'

import {
  View,
  Text,
  Image,
  StatusBar
} from 'react-native'
import { Router, Scene } from 'react-native-router-flux';
import { Provider, connect } from 'react-redux'


import mockCamera from './__mocks__/Camera';
jest.mock('react-native-camera', () => mockCamera);

import App from './App'
import CamCam from './components/CamCam'
import store from './store/manageStore'


describe('App', () => {
  // it('renders without crashing', () => {
  //   const div = document.createElement('div')
  //   reactDom.render(<App />, div)
  // })

  it('renders and match to snapshot', () => {
    const tree = renderer.create(<Provider store={store}><App/></Provider> )
    expect(tree).toMatchSnapshot()
  })
})
