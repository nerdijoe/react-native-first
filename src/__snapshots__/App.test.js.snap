// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders and match to snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "powderblue",
        "height": "25%",
        "opacity": 0,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://descargarclashroyalegratis.net/wp-content/uploads/2016/04/clash-min.jpg",
        }
      }
      style={
        Object {
          "height": 217.24137931034483,
          "width": 413.7931034482759,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#EDB1F1",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "margin": 0,
        "padding": 10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        カード
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        カード2
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        アリーナ
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        カメラ
      </Text>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "white",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "alignItems": "flex-start",
                  "display": "flex",
                  "flexDirection": "column",
                  "flexWrap": "wrap",
                  "marginBottom": 10,
                  "marginLeft": 20,
                  "marginTop": 40,
                  "padding": 20,
                }
              }
              dataSource={
                ListViewDataSource {
                  "items": 6,
                }
              }
              renderRow={[Function]}
              renderScrollComponent={[Function]}
            >
              <View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 8,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "margin": 3,
                      "minWidth": "33.3333%",
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={
                      Object {
                        "uri": "http://www.clashapi.xyz/images/cards/arrows.png",
                      }
                    }
                    style={
                      Object {
                        "height": 131.5,
                        "width": 106,
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 30,
                        "padding": 10,
                      }
                    }
                  >
                    Arrows
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 8,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "margin": 3,
                      "minWidth": "33.3333%",
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={
                      Object {
                        "uri": "http://www.clashapi.xyz/images/cards/bomber.png",
                      }
                    }
                    style={
                      Object {
                        "height": 131.5,
                        "width": 106,
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 30,
                        "padding": 10,
                      }
                    }
                  >
                    Bomber
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 8,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "margin": 3,
                      "minWidth": "33.3333%",
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={
                      Object {
                        "uri": "http://www.clashapi.xyz/images/cards/archers.png",
                      }
                    }
                    style={
                      Object {
                        "height": 131.5,
                        "width": 106,
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 30,
                        "padding": 10,
                      }
                    }
                  >
                    Archers
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 8,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "margin": 3,
                      "minWidth": "33.3333%",
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={
                      Object {
                        "uri": "http://www.clashapi.xyz/images/cards/knight.png",
                      }
                    }
                    style={
                      Object {
                        "height": 131.5,
                        "width": 106,
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 30,
                        "padding": 10,
                      }
                    }
                  >
                    Knight
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 8,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "margin": 3,
                      "minWidth": "33.3333%",
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={
                      Object {
                        "uri": "http://www.clashapi.xyz/images/cards/fireball.png",
                      }
                    }
                    style={
                      Object {
                        "height": 131.5,
                        "width": 106,
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 30,
                        "padding": 10,
                      }
                    }
                  >
                    Fireball
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 8,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "margin": 3,
                      "minWidth": "33.3333%",
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Image
                    source={
                      Object {
                        "uri": "http://www.clashapi.xyz/images/cards/mini-pekka.png",
                      }
                    }
                    style={
                      Object {
                        "height": 131.5,
                        "width": 106,
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 30,
                        "padding": 10,
                      }
                    }
                  >
                    Mini P.E.K.K.A.
                  </Text>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#8EF6E4",
          "borderBottomColor": "#828287",
          "borderBottomWidth": 0.5,
          "height": 64,
          "left": 0,
          "paddingTop": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "left": 0,
              "marginTop": 10,
              "position": "absolute",
              "right": 0,
              "top": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            lineBreakMode="tail"
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "center",
                "color": "#0A0A0A",
                "fontSize": 18,
                "left": 0,
                "opacity": 1,
                "right": 0,
                "textAlign": "center",
                "width": 180,
              }
            }
          >
            Cards with ListView
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "margin": 5,
        "padding": 3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 7,
        }
      }
    >
      © 2017, クラロワ大好きです。
    </Text>
  </View>
</View>
`;
